src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -1429,6 +1429,58 @@ class SelectStatementSegment(ansi.SelectStatementSegment):
     )
 
 
+class LimitClauseSegment(BaseSegment):
+    """A `LIMIT` clause like in `SELECT`."""
+
+    type = "limit_clause"
+    match_grammar: Matchable = Sequence(
+        "LIMIT",
+        Indent,
+        OneOf(
+            OptionallyBracketed(
+                Sequence(
+                    # Allow a number by itself OR
+                    Ref("NumericLiteralSegment"),
+                    # Arithmetic operations followed by a number any number of times
+                    AnyNumberOf(
+                        Sequence(
+                            Ref("ArithmeticBinaryOperatorGrammar"),
+                            Ref("NumericLiteralSegment"),
+                        ),
+                        optional=True,
+                    ),
+                )
+            ),
+            "ALL",
+        ),
+        OneOf(
+            Sequence(
+                "OFFSET",
+                OptionallyBracketed(
+                    Sequence(
+                        # Allow a number by itself OR
+                        Ref("NumericLiteralSegment"),
+                        # Arithmetic operations followed by a number any number of times
+                        AnyNumberOf(
+                            Sequence(
+                                Ref("ArithmeticBinaryOperatorGrammar"),
+                                Ref("NumericLiteralSegment"),
+                            ),
+                            optional=True,
+                        ),
+                    )
```

Comment:
Same comment here as above - that we should allow either a numeric literal or an expression.